components/net/fetch/methods.rs

```diff
@@ -944,7 +947,7 @@ pub fn should_request_be_blocked_as_mixed_content(request: &Request) -> bool {
     }
 
     // 1.2. request’s URL is a potentially trustworthy URL.
-    if request.url().is_potentially_trustworthy() {
+    if protocol_registry.is_url_potentially_trustworthy(&request.url()) {
         return false;
     }
```

Comment:
I would just put `url.is_potentially_trustworthy() || protocol_registry.is_secure(url.scheme())` here and elsewhere to make it clear what the ProtocolRegistry is up to.

Reply:
I have done this mostly because I want to replace as many plain `is_potentially_trustworthy` checks as possible, and it's a bit easier to check this way

Reply:
Passing the protocol registry as an argument looks reasonable to me.